### PR TITLE
plant growth form - scraped data + work from vocab workshop 

### PR DESCRIPTION
**Pull request by ehwenk**
* branch with curated `plant_growth_form` dataset (near complete for all Australian taxa)
* branch with scraped data for a cluster of `plant growth form` - related traits (data from state floras + FOA)
* updated trait definitions + trait mappings for plant growth form related traits, following vocabulary workshop in August/September

### CODE REVIEW

File: config/traits.yml [hunk 1]
```diff
@@ -2255,37 +2265,30 @@ traits:
...
-        hemicryptophyte: Plants with buds at or near the soil surface
-        hemicryptophyte_flat_rosette: Plants with buds at or near the soil surface; all leaves radical, leaves flat or erect depending on conditions
-        hemicryptophyte_erect_rosette: Plants with buds at or near the soil surface; all leaves radical, leaves always erect
-        hemicryptophyte_partial_rosette: Plants with buds at or near the soil surface; radical and cauline leaves present, largest leaves on lower stem
-        hemicryptophyte_proto-rosette: Plants with buds at or near the soil surface; all leaves cauline, largest leaves towards the middle of stem
-        hydrophytes: Plants with bulbs that rest submerged under water
-        phanerophyte: Plants with dominant buds on branches which project freely into the air; normally woody perennials
-        therophyte: Annual plants that complete their lives rapidly in favourable conditions and survive the unfavourable cold or dry season in the form of seeds
+        phanerophyte: Plant whose perennating buds are on persistent shoots which rise at least 50 cm above the soil surface. These are normally woody perennials.
```
[the remark this answers is not in the document]
Comment (dfalster) [reply config/traits.yml:2272]: I was going to write the same for `life_history` below but I see you've sorted those according to lifsetage duration. Whatever the strategy, let's be consistent within definitions file

File: data/FloraLucid_2020/metadata.yml [hunk 2]
```diff
@@ -0,0 +1,331 @@
+source:
+  primary:
+    key: FloraLucid_2020
```
Comment (dfalster) [inline data/FloraLucid_2020/metadata.yml:3]: We'd better rename this. Lucid is not the name of a flora but a platform hosting many id keys. See https://www.lucidcentral.org

So maybe `FloraATRP`?

That said, these names seems inconsistent with other AusTraits entries, e.b. ANBG_2019, SAH_2019, etc

File: data/PlantGrowthFormCurated_2022/metadata.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          plant_growth_form_to_use, "monocot_large_basal", "basal_large"),
+        woodiness_to_use = stringr::str_replace(
+          woodiness_to_use, "soft_woody", "semi_woody"),
+        woodiness_to_use = stringr::str_replace(
+          woodiness_to_use, "woody_stemmed_monocot", "woody_like_stem"),
+      )
+  '
+  collection_date: unknown/2022
+  taxon_name: taxon_name
+  description: A curated collection of plant growth form values that includes all species in the Australian Plant Census (APC).
```
Comment (dfalster) [inline data/PlantGrowthFormCurated_2022/metadata.yml:39]: What's the basis for including the curated dataset as a standalone dataset? ARE duplicates with all other floras? I think it's answered in sampling strategy, but suggest making clear what's included here and why. 

For example, is this dataset all plants in APC (as it says), or just those not in the other sources?

**Review by dfalster — COMMENTED** [review]
This is an epic amount of work. It's very hard for me to review, as I would need a long time with each dataset. Superficially looks good. See questions below regarding naming.

File: data/FloraOfAustralia_2022/metadata.yml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  primary:
+    key: FloraOfAustralia_2022
+    bibtype: Online
+    year: 2022
+    author: Australian Biological Resources Study
+    title: Flora of Australia, Australian Biological Resources Study, Canberra.
+    url: http://www.ausflora.org.au
+contributors:
+  data_collectors:
+  - last_name: Coleman
```
Comment (dfalster) [inline data/FloraOfAustralia_2022/metadata.yml:11]: We'd better follow up with each flora/group and find out who is best to credit here. 

David can come in as last collector. I think we have the option to add notes for authors. In that case, good to specific what David's role was ""E.g. Responsible for extracting data from online resource"

Comment (ehwenk) [reply data/FloraOfAustralia_2022/metadata.yml]: I've added notes below David's name and more importantly removed these datasets from this branch while we contact the floras

File: data/PlantGrowthFormCurated_2022/metadata.yml (hunk 3 above)
Comment (ehwenk) [reply data/PlantGrowthFormCurated_2022/metadata.yml]: There are duplicates, but the `curated` dataset represents a consensus across floras with a significant number of taxa (4000+) being filled in manually and many more where the description was read to confirm the automated score.

File: config/traits.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ehwenk) [reply config/traits.yml:2272]: It is purposeful - For many traits I've sorted the trait values in a way that puts more similar concepts near one another; and often places commonly used values above rarely used values

File: data/FloraLucid_2020/metadata.yml (hunk 2 above)
Comment (ehwenk) [reply data/FloraLucid_2020/metadata.yml]: It is now `ATRP_2020`

**Review by dfalster — APPROVED** [review]

**Pull request merged**